pkg/network/encoding/format.go

```diff
@@ -45,78 +39,109 @@ func (ipc ipCache) Get(addr util.Address) string {
 }
 
 // FormatConnection converts a ConnectionStats into an model.Connection
-func FormatConnection(
-	conn network.ConnectionStats,
-	routes map[string]RouteIdx,
-	httpEncoder *httpEncoder,
-	http2Encoder *http2Encoder,
-	kafkaEncoder *kafkaEncoder,
-	dnsFormatter *dnsFormatter,
-	ipc ipCache,
-	tagsSet *network.TagsSet,
-) *model.Connection {
-	c := connPool.Get().(*model.Connection)
-	c.Pid = int32(conn.Pid)
+func FormatConnection(builder *model.ConnectionBuilder, conn network.ConnectionStats, routes map[string]RouteIdx, httpEncoder *httpEncoder, http2Encoder *http2Encoder, kafkaEncoder *kafkaEncoder, dnsFormatter *dnsFormatter, ipc ipCache, tagsSet *network.TagsSet) {
+
+	builder.SetPid(int32(conn.Pid))
+
 	var containerID string
 	if conn.ContainerID != nil {
 		containerID = *conn.ContainerID
 	}
-	c.Laddr = formatAddr(conn.Source, conn.SPort, containerID, ipc)
-	c.Raddr = formatAddr(conn.Dest, conn.DPort, "", ipc)
-	c.Family = formatFamily(conn.Family)
-	c.Type = formatType(conn.Type)
-	c.IsLocalPortEphemeral = formatEphemeralType(conn.SPortIsEphemeral)
-	c.LastBytesSent = conn.Last.SentBytes
-	c.LastBytesReceived = conn.Last.RecvBytes
-	c.LastPacketsSent = conn.Last.SentPackets
-	c.LastPacketsReceived = conn.Last.RecvPackets
-	c.LastRetransmits = conn.Last.Retransmits
-	c.Direction = formatDirection(conn.Direction)
-	c.NetNS = conn.NetNS
-	c.RemoteNetworkId = ""
-	c.IpTranslation = formatIPTranslation(conn.IPTranslation, ipc)
-	c.Rtt = conn.RTT
-	c.RttVar = conn.RTTVar
-	c.IntraHost = conn.IntraHost
-	c.LastTcpEstablished = conn.Last.TCPEstablished
-	c.LastTcpClosed = conn.Last.TCPClosed
-	c.Protocol = formatProtocolStack(conn.ProtocolStack, conn.StaticTags)
-
-	c.RouteIdx = formatRouteIdx(conn.Via, routes)
-	dnsFormatter.FormatConnectionDNS(conn, c)
+
+	builder.SetLaddr(func(w *model.AddrBuilder) {
+		w.SetIp(ipc.Get(conn.Source))
+		w.SetPort(int32(conn.SPort))
+		w.SetContainerId(containerID)
+	})
+	builder.SetRaddr(func(w *model.AddrBuilder) {
+		w.SetIp(ipc.Get(conn.Dest))
+		w.SetPort(int32(conn.DPort))
+	})
+	builder.SetFamily(uint64(formatFamily(conn.Family)))
+	builder.SetType(uint64(formatType(conn.Type)))
+	builder.SetIsLocalPortEphemeral(uint64(formatEphemeralType(conn.SPortIsEphemeral)))
+	builder.SetLastBytesSent(conn.Last.SentBytes)
+	builder.SetLastBytesReceived(conn.Last.RecvBytes)
+	builder.SetLastRetransmits(conn.Last.Retransmits)
+	builder.SetLastPacketsReceived(conn.Last.RecvPackets)
+	builder.SetDirection(uint64(formatDirection(conn.Direction)))
+	builder.SetNetNS(conn.NetNS)
+	builder.SetRemoteNetworkId("")
```

Comment:
do we need to explicitly set it if it is empty / default value?

Reply:
We shouldn't no